nautobot/core/tests/integration/test_home.py

```diff
@@ -1,17 +1,145 @@
+from django.test.utils import override_settings
+
 from nautobot.utilities.testing.integration import SeleniumTestCase
 
+from nautobot.circuits.models import Circuit, Provider
+from nautobot.dcim.models import PowerFeed, PowerPanel, Site
+from nautobot.tenancy.models import Tenant
 
 class HomeTestCase(SeleniumTestCase):
     """Integration tests against the home page."""
 
     fixtures = ["user-data.json"]  # bob/bob
+    layout = [
+        {
+            "Organization": {
+                "Sites": {
+                    "model": Site,
+                    "permission": "dcim.view_site"
+                },
+                "Tenant": {
+                    "model": Tenant,
+                    "permission": "tenancy.view_tenant"
+                },
+            },
+            "Power": {
+                "Power Feeds": {
+                    "model": PowerFeed,
+                    "permission": "dcim.view_powerfeed"
+                },
+                "Power Panel": {
+                    "model": PowerPanel,
+                    "permission": "dcim.view_powerpanel"
+                },
+            },
+        },
+        {
+            "Circuits": {
+                "Providers": {
+                    "model": Provider,
+                    "permission": "circuits.view_provider"
+                },
+                "Circuits": {
+                    "model": Circuit,
+                    "permission": "circuits.view_circuit"
+                },
+            },
+        },
+    ]
+
+
+    def setUp(self):
+        super().setUp()
+        self.login(self.user.username, self.password)
+
+
+    def tearDown(self):
+        self.logout()
+        super().tearDown()
+
 
     def test_login(self):
         """
         Perform a UI login.
         """
+        # Wait for the page to render and make sure we got a body.
+        self.load_page(self.live_server_url)
 
-        self.login("bob", "bob")
 
-        # Wait for the page to render and make sure we got a body.
-        self.selenium.wait_for_html("body")
+    def test_homepage_render(self):
+        """
+        Render homepage with app defined objects.
+        """
+        # Set test user to admin
+        self.user.is_superuser = True
+        self.user.save()
+
+        self.load_page(self.live_server_url)
+
+        columns_html = self.selenium.find_element_by_class_name("col-sm-6 col-md-4")
```

Comment:
Can we add an additional custom class to these homepage `div`s so that we can look them up using it instead of using these highly generic Bootstrap classes?